2025-09-13 14:22:49,938 - MainThread - sqlmesh.core.config.connection - INFO - Creating new DuckDB adapter for data files: {'md:champions_league_data'} (connection.py:489)
2025-09-13 14:22:50,302 - MainThread - sqlmesh.core.config.connection - INFO - Creating new DuckDB adapter for in-memory database (connection.py:491)
2025-09-13 14:22:50,302 - ThreadPoolExecutor-0_0 - sqlmesh.core.console - WARNING - Model '"champions_league_data"."sqlmesh_example"."full_model"' was not found at /Users/AaronHarkins/Python/SQLMesh_Motherduck/tests/test_full_model.yaml (console.py:2260)
2025-09-13 14:22:50,304 - MainThread - sqlmesh.core.config.connection - INFO - Using existing DuckDB adapter due to overlapping data file: md:champions_league_data (connection.py:479)
2025-09-13 14:22:50,304 - MainThread - sqlmesh.core.console - WARNING - The motherduck engine is not recommended for storing SQLMesh state in production deployments. Please see https://sqlmesh.readthedocs.io/en/stable/guides/configuration/#state-connection for a list of recommended engines and more information. (console.py:2260)
2025-09-13 14:22:56,284 - MainThread - sqlmesh.core.plan.evaluator - INFO - Evaluating plan stage CreateSnapshotRecordsStage (evaluator.py:125)
2025-09-13 14:22:56,733 - MainThread - sqlmesh.core.plan.evaluator - INFO - Evaluating plan stage PhysicalLayerSchemaCreationStage (evaluator.py:125)
2025-09-13 14:22:56,735 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Creating schema 'champions_league_data.sqlmesh__staging' (evaluator.py:1360)
2025-09-13 14:22:56,735 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ CREATE SCHEMA IF NOT EXISTS "champions_league_data"."sqlmesh__staging" (base.py:2466)
2025-09-13 14:22:56,781 - MainThread - sqlmesh.core.plan.evaluator - INFO - Evaluating plan stage BackfillStage (evaluator.py:125)
2025-09-13 14:22:56,784 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Listing data objects in schema champions_league_data.sqlmesh__staging (evaluator.py:1463)
2025-09-13 14:22:56,784 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT CURRENT_CATALOG() (base.py:2466)
2025-09-13 14:22:56,787 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT CURRENT_CATALOG() (base.py:2466)
2025-09-13 14:22:56,790 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT table_name AS name, table_schema AS schema, CASE table_type WHEN 'BASE TABLE' THEN 'table' WHEN 'VIEW' THEN 'view' WHEN 'LOCAL TEMPORARY' THEN 'table' END AS type FROM system.information_schema.tables WHERE (table_catalog = 'champions_league_data' AND table_schema = 'sqlmesh__staging') AND table_name IN ('staging__stg_ucl_squads__1384249225') (base.py:2466)
2025-09-13 14:22:56,813 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Evaluating snapshot SnapshotId<"champions_league_data"."staging"."stg_ucl_squads": 1685672394> (evaluator.py:701)
2025-09-13 14:22:56,814 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Inserting data for snapshot SnapshotId<"champions_league_data"."staging"."stg_ucl_squads": 1685672394> (evaluator.py:910)
2025-09-13 14:22:56,817 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Inserting batch (2025-08-31 00:00:00, 2025-09-07 00:00:00) into champions_league_data.sqlmesh__staging.staging__stg_ucl_squads__1384249225' (evaluator.py:940)
2025-09-13 14:22:56,817 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ DESCRIBE "champions_league_data"."sqlmesh__staging"."staging__stg_ucl_squads__1384249225" (base.py:2466)
2025-09-13 14:22:56,909 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT CURRENT_CATALOG() (base.py:2466)
2025-09-13 14:22:56,909 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT CURRENT_CATALOG() (base.py:2466)
2025-09-13 14:22:56,911 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT table_name AS name, table_schema AS schema, CASE table_type WHEN 'BASE TABLE' THEN 'table' WHEN 'VIEW' THEN 'view' WHEN 'LOCAL TEMPORARY' THEN 'table' END AS type FROM system.information_schema.tables WHERE (table_catalog = 'champions_league_data' AND table_schema = 'sqlmesh__staging') AND table_name IN ('staging__stg_ucl_squads__1384249225') (base.py:2466)
2025-09-13 14:22:56,918 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT CURRENT_CATALOG() (base.py:2466)
2025-09-13 14:22:56,919 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ SELECT type FROM DUCKDB_DATABASES() WHERE database_name = 'champions_league_data' (base.py:2466)
2025-09-13 14:22:56,920 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: /* SQLMESH_PLAN: 0667688c1d624cf4a8fe6fcd7cbb4396 */ CREATE OR REPLACE TABLE "champions_league_data"."sqlmesh__staging"."staging__stg_ucl_squads__1384249225" AS SELECT "team_id" AS "team_id", "team_name" AS "team_name", "id" AS "player_id", "name" AS "player_name", "position" AS "position", "dateofbirth" AS "date_of_birth", "nationality" AS "nationality" FROM "champions_league_data"."main"."champions_league_team_squads" AS "champions_league_team_squads" (base.py:2466)
2025-09-13 14:22:57,004 - MainThread - sqlmesh.core.scheduler - INFO - Execution failed for node EvaluateNode(snapshot_name='"champions_league_data"."staging"."stg_ucl_squads"', interval=(1756598400000, 1757203200000), batch_index=0) (scheduler.py:597)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/utils/concurrency.py", line 234, in sequential_apply_to_dag
    fn(node)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/scheduler.py", line 533, in run_node
    audit_results = self.evaluate(
                    ^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/scheduler.py", line 229, in evaluate
    wap_id = self.snapshot_evaluator.evaluate(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 182, in evaluate
    result = self._evaluate_snapshot(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 797, in _evaluate_snapshot
    self._render_and_insert_snapshot(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 990, in _render_and_insert_snapshot
    apply(query_or_df, index)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 946, in apply
    evaluation_strategy.insert(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 2116, in insert
    self._replace_query_for_model(model, table_name, query_or_df, render_kwargs, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/snapshot/evaluator.py", line 1864, in _replace_query_for_model
    self.adapter.replace_query(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/shared.py", line 312, in internal_wrapper
    return func(*list_args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/base.py", line 512, in replace_query
    return self._create_table_from_source_queries(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/base.py", line 917, in _create_table_from_source_queries
    self._create_table(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/duckdb.py", line 188, in _create_table
    super()._create_table(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/base.py", line 960, in _create_table
    self.execute(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/base.py", line 2443, in execute
    self._execute(sql, track_rows_processed, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/engine_adapter/base.py", line 2475, in _execute
    self.cursor.execute(sql, **kwargs)
duckdb.duckdb.CatalogException: Catalog Error: Table with name champions_league_team_squads does not exist!
Did you mean "champions_league_team_squad"?

LINE 1: ..." AS "date_of_birth", "nationality" AS "nationality" FROM "champions_league_data"."main"."champions_league_team_squads...
                                                                     ^

The above exception was the direct cause of the following exception:

sqlmesh.utils.concurrency.NodeExecutionFailedError: Execution failed for node EvaluateNode(snapshot_name='"champions_league_data"."staging"."stg_ucl_squads"', interval=(1756598400000, 1757203200000), batch_index=0)
2025-09-13 14:22:57,014 - MainThread - sqlmesh.core.context - INFO - Plan application failed. (context.py:1735)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/context.py", line 1727, in apply
    self._apply(plan, circuit_breaker)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/context.py", line 2519, in _apply
    self._scheduler.create_plan_evaluator(self).evaluate(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/plan/evaluator.py", line 107, in evaluate
    self._evaluate_stages(plan_stages, plan)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/plan/evaluator.py", line 127, in _evaluate_stages
    handler(stage, plan)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sqlmesh/core/plan/evaluator.py", line 265, in visit_backfill_stage
    raise PlanError("Plan application failed.")
sqlmesh.utils.errors.PlanError: Plan application failed.
2025-09-13 14:22:57,016 - MainThread - root - INFO - Shutting down the event dispatcher (dispatcher.py:159)
